jmix-gridexport/gridexport-ui/src/main/java/io/jmix/gridexportui/exporter/json/JsonExporter.java

```diff
@@ -63,59 +67,89 @@ public JsonExporter withGsonConfigurator(Function<GsonBuilder, GsonBuilder> gson
-        Collection<Object> items = getItems(table, exportMode);
         Gson gson = createGsonForSerialization();
         JsonArray jsonElements = new JsonArray();
-        for (Object entity : items) {
-            JsonObject jsonObject = new JsonObject();
-            for (Table.Column<Object> column : table.getColumns()) {
-                if (column.getId() instanceof MetaPropertyPath) {
-                    MetaPropertyPath propertyPath = (MetaPropertyPath) column.getId();
-                    Object columnValue = getColumnValue(table, column, entity);
-                    if (columnValue != null) {
-                        jsonObject.add(propertyPath.getMetaProperty().getName(),
-                                new JsonPrimitive(formatValue(columnValue, (MetaPropertyPath) column.getId())));
-                    } else {
-                        jsonObject.add(propertyPath.getMetaProperty().getName(),
-                                JsonNull.INSTANCE);
-                    }
-                } else {
-                    Object columnValue = getColumnValue(table, column, entity);
-                    if (columnValue != null) {
-                        jsonObject.add(column.getStringId(), new JsonPrimitive(formatValue(columnValue)));
-                    }
-                }
+
+        if (exportMode == ExportMode.ALL_ROWS) {
+            jsonAllRecordsExporter.exportAll(table.getItems(),
+                    entity -> addJsonObjectFromTable(table, entity, jsonElements));
+        } else {
+            Collection<Object> items = getItems(table, exportMode);
+
+            for (Object entity : items) {
+                addJsonObjectFromTable(table, entity, jsonElements);
             }
-            jsonElements.add(jsonObject);
         }
+
         downloader.download(new ByteArrayDataProvider(gson.toJson(jsonElements).getBytes(StandardCharsets.UTF_8),
                         uiProperties.getSaveExportedByteArrayDataThresholdBytes(), coreProperties.getTempDir()),
                 getFileName(table) + ".json", DownloadFormat.JSON);
     }
 
     @Override
     public void exportDataGrid(Downloader downloader, DataGrid<Object> dataGrid, ExportMode exportMode) {
-        Collection<Object> items = getItems(dataGrid, exportMode);
         Gson gson = createGsonForSerialization();
         JsonArray jsonElements = new JsonArray();
-        for (Object entity : items) {
-            JsonObject jsonObject = new JsonObject();
-            for (DataGrid.Column<Object> column : dataGrid.getColumns()) {
-                Object columnValue = getColumnValue(dataGrid, column, entity);
-                MetaPropertyPath metaPropertyPath = metadata.getClass(entity).getPropertyPath(column.getId());
+
+        if (exportMode == ExportMode.ALL_ROWS) {
+            jsonAllRecordsExporter.exportAll(dataGrid.getItems(),
+                    entity -> addJsonObjectFromDataGrid(dataGrid, entity, jsonElements));
```

Comment:
Method name `addJsonObjectFromDataGrid(dataGrid, entity, jsonElements)` doesn't look as very "readable": what object is added, where is it added to, etc.

Maybe we can create a method `createJsonObjectFromEntity(Object entity, DataGrid<Object> dataGrid)` that will return JsonObject and then explicitly add it to the jsonElements collection?

```java
JsonObject jsonObject = createJsonObjectFromEntity(Object entity, DataGrid<Object> dataGrid);
jsonElements.add(jsonObject);
```